### PR TITLE
Feat/authenticate with public key

### PR DESCRIPTION
**Pull request by dannil76**

### CODE REVIEW

**Comment by TeddyAndersson** [comment]
Please provide description of how i can test this PR

**Comment by dannil76** [comment]
> Please provide description of how i can test this PR

Ok, I'll ask someone else then... 👯

File: app/libs/authenticate_helper.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import jwt
+from flask import request
+from flask_restful import abort, wraps
+
+def authentication():
```
Comment (TeddyAndersson) [inline app/libs/authenticate_helper.py:5]: Missing functional comment

File: app/libs/authenticate_helper.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    try:
+        jwt.decode(token, public_key, algorithms=['RS256'])
+    except Exception:
+        return False
+
+    return True
+
+
+def authenticate(f):
```
Comment (TeddyAndersson) [inline app/libs/authenticate_helper.py:20]: Authentication and Authenticate is very confusing. Would like some more clear naming for these functions.

Comment (TeddyAndersson) [inline app/libs/authenticate_helper.py:20]: Missing descriptive functional comment

File: app/routes/foo.py [hunk 3]
```diff
@@ -2,12 +2,14 @@
 from flask_restful import Resource
 from marshmallow import ValidationError
 
-from ..libs import parse_application
+from ..libs import parse_application, authenticate
 from ..schemas import ApplicationSchema
 
 
 class Foo(Resource):
```
Comment (TeddyAndersson) [inline app/routes/foo.py:9]: Foo?

**Review by TeddyAndersson — CHANGES_REQUESTED** [review]
Comments for improved readability :)

**Review by TeddyAndersson — APPROVED** [review]
Looks good (y)

**Pull request merged**